spark/src/test/scala/org/apache/spark/sql/delta/DeltaVacuumSuite.scala

```diff
@@ -578,6 +586,182 @@ class DeltaVacuumSuite
+        Row(s"${deltaLog.dataPath}", 300000L, true, 0L)
+      )
+      val inventory = spark.createDataFrame(spark.sparkContext.parallelize(data),
+        VacuumCommand.INVENTORY_SCHEMA)
+      gcTest(deltaLog, clock)(
+        CreateFile("file1.txt", commitToActionLog = true, Map("_underscore_col_" -> "10")),
+        CreateFile("file2.txt", commitToActionLog = false, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test.txt", true, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test2.txt", false, Map("_underscore_col_" -> "10")),
+        CheckFiles(Seq("file1.txt", "_underscore_col_=10")),
+        CheckFiles(Seq("file2.txt", "_underscore_col_=10")),
+        LogicallyDeleteFile("_underscore_col_=10/test.txt"),
+        AdvanceClock(defaultTombstoneInterval + 1000),
+        GCByInventory(dryRun = true, expectedDf = Seq(
+          s"${deltaLog.dataPath}/file2.txt",
+          s"${deltaLog.dataPath}/_underscore_col_=10/test.txt",
+          s"${deltaLog.dataPath}/_underscore_col_=10/test2.txt"
+        ), inventory = Some(inventory)),
+        GCByInventory(dryRun = false, expectedDf = Seq(tempDir), inventory = Some(inventory)),
+        CheckFiles(Seq("file1.txt")),
+        CheckFiles(Seq("file2.txt", "_underscore_col_=10/test.txt",
+          "_underscore_col_=10/test2.txt"), exist = false)
+      )
+    }
+  }
+
+  test("vacuum using inventory delta table and should not touch hidden files") {
+    withSQLConf(DeltaSQLConf.DELTA_VACUUM_RETENTION_CHECK_ENABLED.key -> "false") {
+      withEnvironment { (tempDir, clock) =>
+        import testImplicits._
+        val path = s"""${tempDir.getCanonicalPath}_data"""
+        val inventoryPath = s"""${tempDir.getCanonicalPath}_inventory"""
+
+        // Define test delta table
+        val data = Seq(
+          (10, 1, "a"),
+          (10, 2, "a"),
+          (10, 3, "a"),
+          (10, 4, "a"),
+          (10, 5, "a")
+        )
+        data.toDF("v1", "v2", "v3")
+          .write
+          .partitionBy("v1", "v2")
+          .format("delta")
+          .save(path)
+        val deltaLog = DeltaLog.forTable(spark, path)
+        val reservoirData = Seq(
+          Row(s"${deltaLog.dataPath}/file1.txt", 300000L, false, 0L),
+          Row(s"${deltaLog.dataPath}/file2.txt", 300000L, false, 0L),
+          Row(s"${deltaLog.dataPath}/_underscore_col_=10/test.txt", 300000L, false, 0L),
+          Row(s"${deltaLog.dataPath}/_underscore_col_=10/test2.txt", 300000L, false, 0L)
+        )
+        spark.createDataFrame(
+          spark.sparkContext.parallelize(reservoirData), VacuumCommand.INVENTORY_SCHEMA)
+          .write
+          .format("delta")
+          .save(inventoryPath)
+        gcTest(deltaLog, clock)(
+          CreateFile("file1.txt", commitToActionLog = false),
+          CreateFile("file2.txt", commitToActionLog = false),
+          CreateFile("_underscore_col_=10/test.txt", false),
```

Comment:
Let's call out in a comment that `_underscore_col_` is not a partition column, so then this must be treated as a hidden file.